test/jb_projects/transfer_handle_of.test.js

```diff
@@ -0,0 +1,173 @@
+    await expect(storedProjectId).to.equal(PROJECT_ID_1);
+    await expect(storedOldProjectId).to.equal(0);
+
+    await expect(tx)
+      .to.emit(jbProjectsStore, 'TransferHandle')
+      .withArgs(
+        PROJECT_ID_1,
+        deployer.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+        projectOwner.address,
+      );
+  });
+
+  it(`Can't transfer handle if is empty handle`, async function () {
+    const { projectOwner, deployer, jbProjectsStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    await expect(
+      jbProjectsStore
+        .connect(projectOwner)
+        .transferHandleOf(
+          PROJECT_ID_1,
+          deployer.address,
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_EMPTY),
+        ),
+    ).to.be.revertedWith('0x0a: EMPTY_HANDLE');
+  });
+
+  it(`Can't transfer handle if handle taken already`, async function () {
+    const { projectOwner, deployer, jbProjectsStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    await expect(
+      jbProjectsStore
+        .connect(projectOwner)
+        .transferHandleOf(
+          PROJECT_ID_1,
+          deployer.address,
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        ),
+    ).to.be.revertedWith('0x0b: HANDLE_TAKEN');
+  });
+
+  it(`Can't transfer handle if not owner of project`, async function () {
+    const { projectOwner, deployer, addrs, jbProjectsStore, mockJbOperatorStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    await expect(
+      jbProjectsStore
+        .connect(addrs[0])
+        .transferHandleOf(
+          PROJECT_ID_1,
+          deployer.address,
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        ),
+    ).to.be.reverted;
+  });
+
+  it(`Can't transfer handle if operator even with permissions`, async function () {
+    const { projectOwner, deployer, addrs, jbProjectsStore, mockJbOperatorStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(deployer.address, projectOwner.address, PROJECT_ID_1, SET_HANDLE_PERMISSION_INDEX)
+      .returns(true);
+
+    await expect(
+      jbProjectsStore
+        .connect(deployer)
+        .transferHandleOf(
```

Comment:
this is failing because the `_newHandle` is PROJECT_HANDLE_1 which is the same as the current handle for the project with PROJECT_ID_1. The error here is 0x0b: HANDLE_TAKEN

Reply:
You are right, it is fixed now. thx.